reason-react-tutorial.md

```diff
@@ -0,0 +1,681 @@
+
+Basically, the reason that the final argument has to be unit is to signal when to stop currying the function. Since Reason function arguments are automatically [curried](https://github.com/getify/Functional-Light-JS/blob/13a3bdafb4edb83207db76212312472aab20d06a/manuscript/ch3.md/#one-at-a-time), calling `ReactDOMRe.Style.make(~padding="1rem");` returns _a new function_ that we could pass more CSS properties into. Using functions like this lets us progressively apply values into our function throughout our application rather than all at once.
+
+If we wanted to pull our styles outside of `render`, I've found it helpful to use a _[local module](https://reasonml.github.io/docs/en/module)_. This can help add some readability to our `render` if styles are getting a little long.
+
+{{< highlight javascript >}}
+/* Inside of `Card.re` */
+module Styles = {
+  let card =
+    ReactDOMRe.Style.make(
+      ~border="2px solid #898989",
+      ~borderRadius="4px",
+      ~padding="1rem",
+      (),
+    );
+};
+
+/* Later, in our `render` JSX */
+<div style=Styles.card>
+{{< /highlight >}}
+
+Another commonly-used community solution to styling is `[bs-css](https://github.com/SentiaAnalytics/bs-css)`, which is a typed wrapper around [emotion](https://emotion.sh/). If we wanted to use `bs-css` first we would need to install it.
+
+{{< highlight javascript >}}
+npm install --save bs-css
+{{< /highlight >}}
+
+And then we will need to add `bs-css` to the `"bs-dependencies"` field in our `bsconfig.json` file (if you cloned the sample repo it will be right there alongside `package.json`).
+
+{{< highlight javascript >}}
+"bs-dependencies": ["reason-react", "bs-css"],
+{{< /highlight >}}
+
+Now we can go convert our styles to use `bs-css`, which will generate a string that we can use as a `className`. Using `bs-css` gives a little more type safety to our css styles, if that's something that you're looking for.
+
+{{< highlight javascript >}}
+/* Inside `Card.re` */
+
+module Styles = {
+  open Css;
+  let card = style([
+    border(px(1), `solid, hex("898989")),
+    borderRadius(px(4)),
+    padding(rem(1.0))
+  ])
+};
+
+/* Later, in our `render` JSX */
+<div className={Styles.card}>
+{{< /highlight >}}
+
+Sidenote&mdash;if your IDE is yelling about an unbound module warning after you added `bs-css`, try reloading it or re-opening the file. I use VSCode and I commonly get this error after installing new Reason packages. The reason (no pun intended) has to do with the IDE loading dependencies when a file is first opened and you adding dependencies after the file was opened.
```

Comment:
Maybe add the warning message here, so that people really know which one you are referring too.